echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/AbstractEventNotificationAgent.groovy

```diff
@@ -16,17 +16,23 @@
 
 package com.netflix.spinnaker.echo.notification
 
+import com.fasterxml.jackson.databind.ObjectMapper
 import com.netflix.spinnaker.echo.events.EchoEventListener
 import com.netflix.spinnaker.echo.model.Event
 import com.netflix.spinnaker.echo.services.Front50Service
+import groovy.util.logging.Slf4j
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.beans.factory.annotation.Value
 
+@Slf4j
 abstract class AbstractEventNotificationAgent implements EchoEventListener {
 
   @Autowired
   Front50Service front50Service
 
+  @Autowired
+  ObjectMapper mapper = new ObjectMapper()
```

Comment:
Just wondering, why assign a default as well as autowire?

Reply:
Because tests failed without it being autowired. The preference is to use the autowired one if available, but otherwise just use a new one.

Reply:
I've tried to move towards constructor injection for this exact reason; makes it easier to have tests control what is being injected

Reply:
Removed the new object creation, made it `required = false`, and checked that the object isn't null before using it.